manifests/config/apache.pp

```diff
@@ -105,7 +105,7 @@
   Array[Stdlib::Fqdn] $serveraliases = [],
   Stdlib::Port $server_port = 80,
   Stdlib::Port $server_ssl_port = 443,
-  Stdlib::HTTPUrl $proxy_backend = 'http://localhost:3000/',
+  Pattern['^(https?|unix)://'] $proxy_backend = 'unix:///run/foreman.sock|http://localhost/',
```

Comment:
I was thinking about this version:
```suggestion
  Variant[Stdlib::HTTPUrl, Stdlib::Absolutepath] $proxy_backend = 'http://localhost:3000/',
```

Then later inside the class
```puppet
if $proxy_backend =~ Stdlib::Absolutepath {
  $_proxy_backend = "unix://${proxy_backend}|http://${servername}/"
} else {
  $_proxy_backend = $proxy_backend
}
```

However, that probably doesn't work because you still need the `unix://` version to configure Puma.

Perhaps we still need to roughly take that approach and convert it to the Apache or Puma format as close to the service as possible? Then you would also need to do a similar thing in `config.pp` for `$listen_stream`.

After I finish https://github.com/puma/puma/pull/2362 this will be easier.

Reply:
Let me know what you think about the updated design I pushed